game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/CasualtyOrderOfLossesTestOnBigWorldV3.java

```diff
@@ -0,0 +1,134 @@
+      return createUnit(MARINE, count);
+    }
+
+    Collection<Unit> britishArtillery(final int count) {
+      return createUnit(ARTILLERY, count);
+    }
+
+    private Collection<Unit> createUnit(final UnitType unitType, final int count) {
+      return IntStream.range(0, count)
+          .mapToObj(i -> new TripleAUnit(unitType, BRITISH, data))
+          .collect(Collectors.toSet());
+    }
+  }
+
+  @BeforeEach
+  void clearCache() {
+    CasualtyOrderOfLosses.clearOolCache();
+  }
+
+  @Test
+  void improvedArtillery() {
+    addTech(new ImprovedArtillerySupportAdvance(data));
+    final Collection<Unit> attackingUnits = new ArrayList<>();
+    attackingUnits.addAll(DataFactory.britishTank(1));
+    attackingUnits.addAll(DataFactory.britishArtillery(1));
+    attackingUnits.addAll(DataFactory.britishMarine(1));
+    attackingUnits.addAll(DataFactory.britishMarine(1));
+
+    final List<Unit> result =
+        CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(amphibAssault(attackingUnits));
+
+    assertThat(result, hasSize(4));
+    assertThat(result.get(0).getType(), is(TANK));
+    assertThat(result.get(1).getType(), is(MARINE));
+    assertThat(result.get(2).getType(), is(MARINE));
+    assertThat(
+        result.get(3).getType(),
+        is(ARTILLERY)); // << bug we should pick the artillery second or third
+  }
+
+  private void addTech(final TechAdvance techAdvance) {
+    final var change =
+        ChangeFactory.attachmentPropertyChange(
+            TechAttachment.get(BRITISH), true, techAdvance.getProperty());
+    data.performChange(change);
+  }
+
+  private CasualtyOrderOfLosses.Parameters amphibAssault(final Collection<Unit> amphibUnits) {
+    return CasualtyOrderOfLosses.Parameters.builder()
+        .targetsToPickFrom(amphibUnits)
+        .defending(false)
+        .player(BRITISH)
+        .enemyUnits(List.of()) // << TODO: remove this parameter should not matter
+        .amphibious(true)
+        .amphibiousLandAttackers(amphibUnits)
+        .battlesite(FRANCE)
+        .costs(COST_MAP)
+        .territoryEffects(List.of())
+        .data(data)
+        .build();
+  }
+
+  @Test
+  void amphibAssaultWithoutImprovedArtillery() {
+    final Collection<Unit> attackingUnits = new ArrayList<>();
+    attackingUnits.addAll(DataFactory.britishTank(1));
+    attackingUnits.addAll(DataFactory.britishArtillery(1));
+    attackingUnits.addAll(DataFactory.britishMarine(1));
+    attackingUnits.addAll(DataFactory.britishMarine(1));
+
+    final List<Unit> result =
+        CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(amphibAssault(attackingUnits));
+
+    assertThat(result, hasSize(4));
+    assertThat(result.get(0).getType(), is(TANK)); // << bug, should be marine or artillery first
+    assertThat(result.get(1).getType(), is(MARINE)); // << bug should be artillery
+    assertThat(result.get(2).getType(), is(MARINE));
+    assertThat(result.get(3).getType(), is(ARTILLERY)); // << bug, should be tank
```

Comment:
Worth noting, was surprising to see OOL taken in this order. Am I doing the math wrong perhaps to expect tank to be last in this case?